js/modules/k6/experimental/fs/file.go

```diff
@@ -48,7 +51,7 @@ func (f *file) Read(into []byte) (n int, err error) {
 
 	// Check if we have reached the end of the file
 	if currentOffset == fileSize {
-		return 0, newFsError(EOFError, "EOF")
+		return 0, io.EOF
```

Comment:
I might be wrong since I am missing a module context, but this seems like a breaking change. Since later on [there is logic that depends on this handling](https://github.com/grafana/k6/blob/master/js/modules/k6/experimental/fs/module.go#L242-L246) this error type.

Reply:
For this specific case, it isn't: this is the "private" file construct and, as far as I can tell, is never exposed directly to the JS runtime. By the looks of it, we have also updated all the places calling this code to expect `io.EOF` instead of an fsError too 🙇🏻

Reply:
Sorry, I wasn't clear; I meant that the breaking change is that previously, the read method in case of the EOF [resolves with null](https://github.com/grafana/k6/blob/master/js/modules/k6/experimental/fs/module.go#L243), whenever after the changes, it's probably resolved with [the EOF error](https://github.com/grafana/k6/blob/master/js/modules/k6/experimental/fs/module.go#L238).

Reply:
Anyway, keeping in mind that both modules are experimental I'm postponing for you @oleiade decision whenever it should be fixed, or when it should be fixed. The rest of PR looks good to me :+1:

Reply:
Oh, I see what you mean. Thanks for clarifying, indeed 👍🏻 

I reverified, and we're safe —no breaking changes— but I admit the code does not convey what happens clearly enough. 

The module's `Read` method callback checks if the returned error from `File.Read` is `io.EOF`, and if it is, it rewraps it as a `FsError` (which we do expose to the runtime) with its kind set to `EOFEror`. At a later point in the code, if the error is an `FsError` with its kind set to `EOFError`, then it returns null. So the behavior is unchanged.

We have a test to assert that already, but I'm taking note for the cooldown period and see if I can improve it and make it easier to read and understand 🙇🏻